### ISSUE TITLE
Incorrect behavior when connecting a USB sound device (T10582)

### ISSUE THREAD

**Issue by celticmagic** [getsolus/packages#294]
<div dir="rtl"><b>Linky&nbsp;Wolfka</b>&nbsp;(<em>&#35;Linky</em>),&nbsp;2023-07-13&nbsp;07:58:27&nbsp;UTC</div><br/>

# Summary 

When I plug in a USB sound device that can be recognized as a sound output device (a speaker or microphone with a sound output, for example). It is automatically selected as a sound output device, but the sound continues to be output from another device.

## Steps to reproduce
{F10937617}
1Have a 3.5mm jack speaker connected to the PC
2Choose this speaker as the sound output device.
3Connect a USB microphone with sound output capability (recognized as an input AND output device).
4 The audio output device is now the microphone, but sound continues to come out of the speaker.

## Expected result

The expected result (the behavior I had on other distributions):
When I plug in the microphone the sound continues to come out through the loudspeaker and in the settings the loudspeaker is always selected.

Another expected result would be that if my microphone is considered as the sound output device to be used, the sound should stop coming out of the speaker without needing to deselect it and select it again (like when plugging in a headset).

## Actual result

The sound is not coming out of the right device.

## Environment

  - Is system up to date?  yes  
  - Repo: Shannon  
  - Desktop Environment : Budgie

## System details
{F10937613}

## Other comments 
My main problem is that without choosing my speaker again, even though the sound comes out of it, I can't adjust the volume.
I can do the same thing as shown in the video and get the same result over and over again.
I'm not sure whether this problem occurs when I plug in the microphone before turning on the PC.

**Comment by TraceyC77** [comment]
I tested this with a SteelSeries wireless headset with USB base station. It has a mic, so it has both output and input.
I was unable to reproduce the bug and can confirm that sound works as expected. Closing this as resolved. 

Test steps:
1. Make sure headset is disconnected, USB cable disconnected from base station. Output audio device set to "Built-in Audio Pro".
2. Restart Firefox, start playing a YouTube video. Verified in the sound tray widget it was playing through "Built-in Audio Pro".
3. Connect the SteelSeries base station to USB input, turn on headset.

Expected result:
- Sound comes out through the device indicated by the sound system tray icon.
- Selected audio device is shown as "Built-in Audio Pro" until I change it.
- The audio from Firefox plays through the device selected via the sound icon.
- When switching to the headset via the system tray audio icon, the sound from Firefox switches to the selected device.
